tests/testthat/test-harmonisation.R

```diff
@@ -1,19 +1,330 @@
 # Need extensive testing of different harmonisation scenarios
```

Comment:
Thanks for putting in these tests they're really useful. I suggest refactoring the code below to make it a bit more succinct, the code is almost identical for each scenario

```
# Need extensive testing of different harmonisation scenarios
library(dplyr)

scenarios <- as_tibble(read.csv("harmonisation_scenarios.csv"))

# Get all scenarios, remote flip2 for now
scen <- unique(scenarios$scenario)
scen <- scen[!scen %in% c("flip2")]
# test each scenario 1 by 1
# - Create dataset of input
# - Extract harmonised data
# - Compare against truth
lapply(scen, function(s)
{
  test_that(s,
  {
    x <- subset(scenarios, scenario == s)
    df <- lapply(unique(x$id), \(i) {
      scenarios %>% filter(scenario == s, id == i, version == "input")
    })
    action <- ifelse(s %in% c("palindrome_flip"), 2, 1)
    dataset <- create_dataset(data=df, harmonise = TRUE, tolerance = 0.08, action = action) %>% 
      suppressMessages()
    result <- lapply(1:getLength(dataset), function(i) {
      getSummarySet(dataset, i) %>%
        getSummaryData() %>%
        select(chr, ea, nea, eaf, beta)
    })
    truth <- lapply(unique(x$id), \(i) {
      scenarios %>% filter(scenario == s, id == i, version == "truth") %>% 
        select(chr, ea, nea, eaf, beta)
    })
    expect_equal(truth, result)
  })
})


# TODO

# test_that("harmonisation scenario: flip2", {
# test <- "flip2"

# scenarios <- as_tibble(read.csv("harmonisation_scenarios.csv"))
# ids <- 1:2


# df <- lapply(seq_along(ids), function(i) {
#   d <- scenarios %>% filter(scenario == test, id == ids[i],version == "input")})

# dataset <- create_dataset(data=df, harmonise = TRUE, tolerance = 0.08, action = 2)


# result <- lapply(seq_along(dataset@summary_sets), function(i) {
#   d <- dataset@summary_sets[[i]]@ss %>% select(chr, ea, nea, eaf, beta)})

# # Truth
# truth<- lapply(seq_along(ids), function(i) {
#   d <- scenarios %>% filter(scenario == test, id == ids[i],version == "truth")  %>% 
#     select(chr, ea, nea, eaf, beta)})


# lapply(seq_along(ids), function(i) {
#   expect_equal(truth[i], result[i])})

# })
```

Reply:
I had the read_tibble outside the tests, but there was some problem with the check() analyses and that is why it is like this. But I will look at it again.